launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -410,7 +410,7 @@ public SparkAppHandle startApplication(SparkAppHandle.Listener... listeners) thr
     return handle;
   }
 
-  private ProcessBuilder createBuilder() throws IOException {
```

Comment:
I think you can just reflect this method and call `sparkLauncher.createBuilder().command()` as described in the mailing list. Not sure how much useful this API is in general yet.

Reply:
How about simply logging the command from `SparkLauncher` instead of adding a new public method, then?

Reply:
How is it differernt from setting `SPARK_PRINT_LAUNCH_COMMAND` to 1?

Reply:
That prints the command to stderr, which is not always easy to access (depending on the environment, etc.). Using the logging API gives clients more flexibility in controlling where the information appears (they can use the full power of log4j, for example).